src/txHashBuilder.c

```diff
@@ -862,9 +887,88 @@ static void txHashBuilder_assertCanLeaveWithdrawals(tx_hash_builder_t* builder)
 	ASSERT(builder->remainingWithdrawals == 0);
 }
 
-void txHashBuilder_addAuxData(tx_hash_builder_t* builder, const uint8_t* auxDataHashBuffer, size_t auxDataHashBufferSize)
+void txHashBuilder_enterMint(tx_hash_builder_t* builder)
 {
 	txHashBuilder_assertCanLeaveWithdrawals(builder);
+	{
+		// Enter mint
+		BUILDER_APPEND_CBOR(CBOR_TYPE_UNSIGNED, TX_BODY_KEY_MINT);
+	}
+	builder->state = TX_HASH_BUILDER_IN_MINT;
+}
+
+void txHashBuilder_addMint_topLevelData(
+        tx_hash_builder_t* builder, uint16_t numAssetGroups
+)
+{
+	_TRACE("state = %u", builder->state);
+
+	ASSERT(builder->state == TX_HASH_BUILDER_IN_MINT);
+
+	builder->outputData.remainingAssetGroups = numAssetGroups;
+	// Map(numAssetGroups)[
+	//   { * policy_id => { * asset_name => uint } }
+	// ]
+	BUILDER_APPEND_CBOR(CBOR_TYPE_MAP, numAssetGroups);
+	if (0 != numAssetGroups) {
+		builder->state = TX_HASH_BUILDER_IN_MINT_ASSET_GROUP;
+	} else {
+		// TODO is this an error case?
+	}
+}
+
+void txHashBuilder_addMint_tokenGroup(
+        tx_hash_builder_t* builder,
+        const uint8_t* policyIdBuffer, size_t policyIdSize,
+        uint16_t numTokens
+)
+{
+	addTokenGroup(builder, policyIdBuffer, policyIdSize, numTokens,
+	              TX_HASH_BUILDER_IN_MINT_ASSET_GROUP, TX_HASH_BUILDER_IN_MINT_TOKEN);
+}
+
+void txHashBuilder_addMint_token(
+        tx_hash_builder_t* builder,
+        const uint8_t* assetNameBuffer, size_t assetNameSize,
+        int64_t amount
+)
+{
+	//  TODO int64 -> uint64 conversion should work fine ?
```

Comment:
Unfortunately not, if the int64 is too big (cannot be represented by uint64). There is no simple type that could hold both. The only way around it would be to have a union of the two plus one byte for an enum to distinguish how to interpret it, but that sounds like an overkill (I would not do it just for the purpose of txhashbuilder, but having `multiasset_amount_t` might actually be useful if we wanted to have common APDU handlers and/or ui state machines).

Also, one cannot use
`BUILDER_APPEND_CBOR(CBOR_TYPE_UNSIGNED, amount);` with int64.

Please consult with @balintklement  that ledgerjs should contain tests with maximum- and minimum-valued integers in multiasset outputs and mints.